tests/operator/helper.go

```diff
@@ -45,3 +57,62 @@ func SplitCsv(csv string) CsvResult {
 	}
 	return result
 }
+
+func hasOperatorInstallModeSingleNamespace(installModes []v1alpha1.InstallMode) bool {
+	for i := 0; i < len(installModes); i++ {
+		if installModes[i].Type == v1alpha1.InstallModeTypeSingleNamespace && installModes[i].Supported {
+			return true
+		}
+	}
+	return false
+}
+
+func filterSingleNamespacedOperatorUnderTest(operators []*provider.Operator) (singleNamespacedOperators []*provider.Operator) {
+	for _, operator := range operators {
+		if hasOperatorInstallModeSingleNamespace(operator.Csv.Spec.InstallModes) && len(operator.TargetNamespaces) == 1 {
+			singleNamespacedOperators = append(singleNamespacedOperators, operator)
+		}
+	}
+	return singleNamespacedOperators
+}
+
+// This function checks if the namespace contains only valid operator pods
+func checkIfNamespaceContainsOnlyOperatorPods(namespace string) (isOperatorOnlyNamespace bool, err error) {
+	// Get all pods from the target namespace
+	podsList, err := clientsholder.GetClientsHolder().K8sClient.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{})
```

Comment:
why not use the env->Pods list instead and filter the namespace?

Reply:
That's because there could be some pods created by others running in the operator namespace which are not test pods. env.Pods would give only test pods, but we need to check for all pods in a namespace.

Reply:
Should we really test all pods? If some pods need to be skipped because they do not belong to the cnf under test (like some specific cluster scenario), we can currently skip them by setting labels.  This looks like a similar scenario to me. By default, if there is no labels configured for pods, all pods in the namespace are considered anyways.

Reply:
We also already gather all pods in `env.AllPods` and we could just filter the namespace as well for David's first comment.